src/extractor/guidance/turn_analysis.cpp

```diff
@@ -117,14 +117,22 @@ std::vector<TurnOperation> TurnAnalysis::getTurns(const NodeID from, const EdgeI
         {
             can_enter_roundabout = true;
         }
-        else
+        // Exiting roundabouts at an entry point is technically a data-modelling issue.
+        // This workaround handles cases in which an exit follows the entry.
+        // To correctly represent perceived exits, we only count exits leading to a
+        // separate vertex than the one we are coming from that are in the direction of
+        // the roundabout.
+        // The sorting of the angles represents a problem for left-sided driving, though.
+        // FIXME in case of left-sided driving, we have to check whether we can enter the
+        // roundabout later in the cycle, rather than prior.
```

Comment:
I have no background on how this is handled currently. I just know that the decision whether an exit exists past the point of entry depends on the direction in which the roundabout can be entered. If an exit is between entry and origin, we need to count it. This `between` depends on whether we drive clockwise or counterclockwise in the roundabout